cpp/src/gandiva/CMakeLists.txt

```diff
@@ -27,16 +27,7 @@ add_dependencies(gandiva-all gandiva gandiva-tests gandiva-benchmarks)
 
 find_package(LLVMAlt REQUIRED)
 
-if(LLVM_VERSION_MAJOR LESS "10")
-  set(GANDIVA_CXX_STANDARD ${CMAKE_CXX_STANDARD})
-else()
-  # LLVM 10 or later requires C++ 14
-  if(CMAKE_CXX_STANDARD LESS 14)
-    set(GANDIVA_CXX_STANDARD 14)
-  else()
-    set(GANDIVA_CXX_STANDARD ${CMAKE_CXX_STANDARD})
-  endif()
-endif()
+set(GANDIVA_CXX_STANDARD ${CMAKE_CXX_STANDARD})
```

Comment:
How about removing `GANDIVA_CXX_STANDARD` and using `CMAKE_CXX_STANDARD` directly?